File: Source/Aladdin.CAPI.STB.Java/src/aladdin/asn1/stb/GOSTSBlock.java
package aladdin.asn1.stb;
import aladdin.asn1.*; 
import java.io.*; 

////////////////////////////////////////////////////////////////////////////////
// GOSTParams ::= SEQUENCE {
// 	sblock SBlock OPTIONAL
// }
////////////////////////////////////////////////////////////////////////////////
public final class GOSTSBlock extends Sequence<IEncodable>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ChoiceCreator(SBlock.class).factory(), Cast.O), 
	}; 
	// конструктор при раскодировании
	public GOSTSBlock(IEncodable encodable) throws IOException { super(encodable, info); } 
    
	// конструктор при закодировании
	public GOSTSBlock(OctetString sblock) { super(info, sblock); }
    
	// конструктор при закодировании
	public GOSTSBlock(ObjectIdentifier sblock) { super(info, sblock); }
    
	public final IEncodable	sblock() { return get(0); }
}

File: Source/Aladdin.CAPI.STB/ASN1/STB/GOSTSBlock.cs
namespace Aladdin.ASN1.STB
{
    ////////////////////////////////////////////////////////////////////////////////
    // GOSTSBlock ::= SEQUENCE {
    // 	sblock SBlock OPTIONAL
    // }
    ////////////////////////////////////////////////////////////////////////////////
    public class GOSTSBlock : Sequence
    {
	    // информация о структуре
	    private static readonly ObjectInfo[] info = new ObjectInfo[] { 

		    new ObjectInfo(new ChoiceCreator<SBlock>().Factory(), Cast.O), 
	    }; 
	    // конструктор при раскодировании
	    public GOSTSBlock(IEncodable encodable) : base(encodable, info) {} 
    
	    // конструктор при закодировании
	    public GOSTSBlock(OctetString sblock) : base(info, sblock) {}
    
	    // конструктор при закодировании
	    public GOSTSBlock(ObjectIdentifier sblock) : base(info, sblock) {}
    
	    public IEncodable SBlock { get { return this[0]; }}
    }
}
